### PR TITLE
Christmas Event Updates

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: During Christmas you will be able to click on the tree in the Chapel to receive one present per round. That present can contain any item in the game.
/:cl:

Inspired by what goof did for baystation years ago.

Also removed the tree revenge event (dont want to destroy the present spawner) and lowered the santa coming to town event chance so not every single round for a week has a wizard in it.

### CODE REVIEW

**Comment by Nilons** [comment]
this is good

**Comment by BeeSting12** [comment]
nice trips 👍

**Comment by KorPhaeron** [comment]
I didn't test this yet I will soon(TM) if the PR isn't closed for being terrible

**Comment by Potato-Masher** [comment]
> any item in the game

HOOOOOO boy. Now I want to enhance my HoS experience by getting a Holoparasite injector, and use my stand to crush Syndicate scum.

**Comment by BeeSting12** [comment]
If it's literally any item then you'll probably get something utterly useless.

**Comment by Buggy123** [comment]
You could get a pulse rifle!
Or a burnt cigarette butt.

**Comment by Supermichael777** [comment]
like real secret santa

Wait till someone gets a nuke core

**Comment by WJohn** [comment]
You might also end up getting a item that you can't drop, like a rig suit helmet.

**Comment by Buggy123** [comment]
Its all part of the memery, merge posthaste.

**Comment by KorPhaeron** [comment]
Alright I tested it, it all works

**Comment by Hatterhat** [comment]
>any item
wait til a nibba whomst grabbeth an lmg

**Comment by Okand37** [comment]
Instagib rifles, spooky ghost bedsheets, shielded deathsquad hardsuits, bluespace emags, super whetstones, energy axes, chronoguns, and spellbooks? What could go wrong!

I will laugh when someone gets a nuke core and it irradiates them to death, though.

**Comment by Buggy123** [comment]
but christmas

**Comment by arsserpentarium** [comment]
There was idea. Christmas game mode:crew must protect tree from antag grinches.
Possibly grinches can be on pirate ship with same as pirates equipment.
They must unwrench and steal tree.

**Comment by MrMaclang** [comment]
>any item
Pulse rifle.

**Comment by Wutnold124** [comment]
can't wait to unbox a horse dildo that was added in as a joke then forgot about after 20 years

File: code/game/objects/items/gift.dm [hunk 1]
```diff
@@ -72,10 +83,20 @@
...
-	var/obj/item/I = new gift_type(M)
-	M.put_in_hands(I)
-	I.add_fingerprint(M)
+
+/obj/item/a_gift/anything
+	name = "christmas gift"
+	desc = "It could be anything!"
+
+/obj/item/a_gift/anything/get_gift_type()
+	var/list/gift_types_list = subtypesof(/obj/item)
```
Comment (AnturK) [inline code/game/objects/items/gift.dm]: Pretty sure we have this cached somewhere already and if not cache the filtered one.

File: code/game/objects/items/gift.dm [hunk 2]
```diff
@@ -72,10 +83,20 @@
...
+	for(var/V in gift_types_list)
+		if(ispath(V) && ispath(V, /obj/item))
+			var/obj/item/I = V
+			if((!initial(I.icon_state)) || (!initial(I.item_state)) || (initial(I.flags_1) & ABSTRACT_1))
```
Comment (AnturK) [inline code/game/objects/items/gift.dm]: Check DROPDEL too

**Review by AnturK — CHANGES_REQUESTED** [review]

**Comment by davethwave** [comment]
Be traitor on Sybil, have get nuke disk objective. Open present and gets disk but doesn't get greentext since the disk came from bagil.

**Comment by calzilla1** [comment]
Make a message pop up saying some sort of festive greeting from:random player/corperation/meme to:player name

**Comment by KorPhaeron** [comment]
I think I hit an incorrect button

**Comment by Potato-Masher** [comment]
Oof.

**Comment by Buggy123** [comment]
Why tho
reopen

**Comment by davethwave** [comment]
look here
https://github.com/tgstation/tgstation/pull/33690

**Pull request closed without merging**